eda/src/autogluon/eda/analysis/shift.py

```diff
@@ -0,0 +1,347 @@
+from typing import Union, List, Any, Optional
+import warnings
+import pandas as pd
+import numpy as np
+from autogluon.core.metrics import balanced_accuracy, BINARY_METRICS
+from autogluon.tabular import TabularPredictor
+from .. import AnalysisState
+from .base import AbstractAnalysis
+
+
+def post_fit(func):
+    """decorator for post-fit methods"""
+
+    def pff_wrapper(self, *args, **kwargs):
+        assert self._is_fit, f'.fit needs to be called prior to .{func.__name__}'
+        return func(self, *args, **kwargs)
+
+    return pff_wrapper
+
+class Classifier2ST:
+    """A classifier 2 sample test, which tests for a difference between a source and target dataset.  It fits a
+    classifier to predict if a sample is in the source and target dataset, then computes an evaluation metric on a
+    holdout which becomes the test statistic.
+
+    Parameters
+    ----------
+    classifier_class : an AutoGluon predictor, such as autogluon.tabular.TabularPredictor
+        The predictor (classifier) class to classify the source from target dataset, predictor class needs to support
+        binary classification
+    sample_label : str, default = 'xshift_label'
+        The label that will be used to indicate if the sample is from training or test
+    eval_metric : callable, default = autogluon.core.metrics.balanced_accuracy
+        Binary classification metric to use for the classifier 2 sample test
+    split : float, default = 0.5
+        Training/test split proportion for classifier 2 sample test
+    classifier_kwargs : dict, default = {}
+        The kwargs passed to the classifier, a member of classifier_class
+    """
+    def __init__(self,
+                 classifier_class,
+                 sample_label='xshift_label',
+                 eval_metric=balanced_accuracy,
+                 split=0.5,
+                 compute_fi = True,
+                 classifier_kwargs = {}
+                 ):
+        classifier_kwargs.update({'label': sample_label, 'eval_metric': eval_metric})
+        self.classifier = classifier_class(**classifier_kwargs)
+        self.classifier_class = classifier_class
+        self.split = split
+        self.sample_label = sample_label
+        self.eval_metric = eval_metric
+        self._is_fit = False
+        self._test = None
+        self.test_stat = None
+        self.has_fi = None
+        self.compute_fi = compute_fi
+
+    @staticmethod
+    def _make_source_target_label(data, sample_label):
+        """Turn a source, target pair into a single dataframe with label column"""
+        source, target = data[0].copy(), data[1].copy()
+        source.loc[:,sample_label] = 0
+        target.loc[:,sample_label] = 1
+        data = pd.concat((source, target))
```

Comment:
`data = pd.concat((source, target), ignore_index=True)` to avoid overlapping/duplicate indices.